docs/src/main/sphinx/develop/supporting-merge.rst

```diff
@@ -0,0 +1,389 @@
+  from the merge case ``RowBlock``.
+
+  The job of ``storeMergedRows()`` is iterate over the rows in the page,
+  and process them based on the value of the operation column, ``INSERT``,
+  ``DELETE``, ``UPDATE``, or ignore the row.  For some connectors, the
+  ``UPDATE`` operations is transformed into ``DELETE`` and ``INSERT``
+  operations.
+
+* ``finish``::
+
+    ``CompletableFuture<Collection<Slice>> finish()``
+
+  The Trino engine calls ``finish()`` when all the data has been processed by
+  a specific ``ConnectorMergeSink`` instance.  The connector returns a future
+  containing a collection of ``Slice``, representing connector-specific
+  information about the rows processed.  Usually this includes the row count,
+  and might include information like the files or partitions created or
+  changed.
+
+``ConnectorMetadata`` ``MERGE`` API
+-----------------------------------
+
+A connector implementing ``MERGE`` must implement these ``ConnectorMetadata``
+methods.
+
+* ``getRowChangeParadigm()``::
+
+    RowChangeParadigm getRowChangeParadigm(
+        ConnectorSession session,
+        ConnectorTableHandle tableHandle)
+
+  This method is called as the engine starts processing a ``MERGE`` statement.
+  The connector must return a ``RowChangeParadigm`` enum instance.  If the
+  connector does not support ``MERGE`` it should throw a ``NOT_SUPPORTED``
+  exception, meaning that ``SQL MERGE`` is not supported by the connector.
+
+* ``getMergeRowIdColumnHandle()``::
+
+    ColumnHandle getMergeRowIdColumnHandle(
+        ConnectorSession session,
+        ConnectorTableHandle tableHandle,
+        MergeDetails mergeDetails)
+
+  This method is called in the early stages of query planning for ``MERGE``
+  statements.  The ColumnHandle returned provides the ``rowId`` used by the
+  connector to identify rows to be merged, as well as any other fields of
+  the row that the connector needs to complete the ``MERGE`` operation.
+
+* ``getWriteRedistributionColumns()``::
+
+    List<ColumnHandle> getWriteRedistributionColumns(
+        ConnectorSession session,
+        ConnectorTableHandle table)
+
+  This method returns a list of ``ColumnHandles`` for table columns that
+  impact write redistribution, e.g., columns that impact partitioning or
+  bucketing.  By default, this method returns an empty list.
+
+* ``beginMerge()``::
+
+    ConnectorMergeTableHandle beginMerge(
+         ConnectorSession session,
+         ConnectorTableHandle tableHandle,
+         MergeDetails mergeDetails)
+
+  As the last step in creating the ``MERGE`` execution plan, the connector's
+  ``beginMerge()`` method is called, passing the ``session``, the
+  ``tableHandle`` and the ``MergeDetails`` object.
+
+  ``beginMerge()`` performs any orchestration needed in the connector to
+  start processing the ``MERGE``.  This orchestration varies from connector
+  to connector.  In the Hive ACID connector, for example, ``beginMerge()``
```

Comment:
"Hive ACID" is not a connector. 
In the case of Hive connector operating on ACID tables" ?

Reply:
Changed.

Reply:
Changed.